src/wcs2_0/src/main/java/org/geoserver/wcs2_0/ScalingPolicy.java

```diff
@@ -82,19 +82,27 @@ public GridCoverage2D scale(
 
             // get scale factor
             double[] scaleFactors = getScaleFactors(scaling);
-            double scaleFactor = scaleFactors[0];
-            scaleFactor = arrangeScaleFactors(hints, new double[] {scaleFactor, scaleFactor})[0];
+            // reading the data can cause the coverage to have asymmetric pre-applied scale factors
+            // due to small numerical differences in their values, so keep both
+            scaleFactors =
+                    arrangeScaleFactors(hints, new double[] {scaleFactors[0], scaleFactors[0]});
 
             // checks
-            if (scaleFactor <= 0) {
+            if (scaleFactors[0] <= 0) {
                 throw new WCS20Exception(
                         "Invalid scale factor",
                         WCS20Exception.WCS20ExceptionCode.InvalidScaleFactor,
-                        String.valueOf(scaleFactor));
+                        String.valueOf(scaleFactors[0]));
+            }
+            if (scaleFactors[1] <= 0) {
+                throw new WCS20Exception(
+                        "Invalid scale factor",
+                        WCS20Exception.WCS20ExceptionCode.InvalidScaleFactor,
+                        String.valueOf(scaleFactors[0]));
             }
 
             // return coverage unchanged if we don't scale
-            if (scaleFactor == 1) {
+            if (scaleFactors[0] == 1 && scaleFactors[1] == 0) {
```

Comment:
sorry for the silly question: is "scaleFactors[1] == 0" correct?
was wondering if that should be == 1

Reply:
You are indeed right, thanks for catching this!